mne/decoding/tests/test_gsoc.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import os.path as op
+from numpy.testing import (assert_array_equal, assert_equal,
+                          assert_array_almost_equal)
+from nose.tools import assert_raises
+from mne import (io, Epochs, read_events, pick_types,
+                compute_raw_covariance)
+from mne.decoding.gsoc import (_EpochsTransformerMixin,
+                               UnsupervisedSpatialFilter,
+                               XdawnTransformer, Vectorizer)
+from mne.decoding.transformer import EpochsVectorizer
+from mne.utils import run_tests_if_main, requires_sklearn
+
+base_dir = op.join(op.dirname(__file__), '..', '..', 'io', 'tests', 'data')
+raw_fname = op.join(base_dir, 'test_raw.fif')
+event_name = op.join(base_dir, 'test-eve.fif')
+
+tmin, tmax = -0.1, 0.2
+event_id = dict(cond2=2, cond3=3)
+
+
+def _get_data():
+    raw = io.read_raw_fif(raw_fname, add_eeg_ref=False, verbose=False,
+                          preload=True)
+    events = read_events(event_name)
+    picks = pick_types(raw.info, meg=False, eeg=True, stim=False,
+                       ecg=False, eog=False,
+                       exclude='bads')[::8]
+    return raw, events, picks
+
+
+def test_epochs_transformer_mixin():
+    raw, events, picks = _get_data()
+    epochs = Epochs(raw, events, event_id, tmin, tmax, preload=True,
+                    baseline=None, verbose=False)
+    ch_names = [epochs.ch_names[p] for p in picks]
+    epochs.pick_channels(ch_names)
+
+    # Test _rehsape method wrong input
+    etm = _EpochsTransformerMixin(n_chan=epochs.info['nchan'])
+    assert_raises(ValueError, etm._reshape, raw)
+
+    # Test _reshape correctness
+    X, y = EpochsVectorizer().fit_transform(epochs)
+    assert_array_equal(etm._reshape(X), epochs._data)
+    assert_equal(etm._reshape(X).ndim, epochs._data.ndim)
+
+
+@requires_sklearn
+def test_unsupervised_spatial_filter():
+    from sklearn.decomposition import PCA
+    raw, events, picks = _get_data()
+    epochs = Epochs(raw, events, event_id, tmin, tmax, picks=picks,
+                    preload=True, baseline=None, verbose=False)
+
+    # Test fit
+    X, y = EpochsVectorizer().fit_transform(epochs)
+    usf = UnsupervisedSpatialFilter(PCA(5), n_chan=epochs.info['nchan'])
+    usf.fit(X)
+    usf1 = UnsupervisedSpatialFilter(PCA(5), n_chan=epochs.info['nchan'])
+
+    # test transform
+    assert_equal(usf.transform(X).ndim, 3)
+
+    # test fit_trasnform
+    print(usf.transform(X), usf1.fit_transform(X))
```

Comment:
can we guess what should be printed? or is it only for making sure repr works? if so add comment so no one is getting too curious what this will print :)